frb_codegen/src/generator/dart/ty_struct.rs

```diff
@@ -120,18 +125,26 @@ impl TypeDartGeneratorTrait for TypeStructRefGenerator<'_> {
                 .collect::<Vec<_>>();
             if has_methods {
                 constructor_params.insert(0, extra_argument);
+                if self.ir.is_exception {
+                    constructor_params.push("Backtrace? backtrace".to_string());
+                }
             }
             let constructor_params = constructor_params.join("");
 
             format!(
-                "{}{}class {} with _${} {{
+                "{}{}class {} with _${} {} {{
                 const factory {}({{{}}}) = _{};
                 {}
             }}",
                 comments,
                 metadata,
                 self.ir.name,
                 self.ir.name,
+                if self.ir.is_exception {
+                    "implements Exception"
+                } else {
+                    ""
+                },
```

Comment:
This `if is_exception "implements Exception"` has appeared at least twice. Maybe we can extract it to a common function, e.g. `impl Ir { fn dart_implements() { if self.is_exception { "implements FrbException" } else "" } }`

Reply:
Requires implementing on the macro `type_dart_generator_struct!(TypeStructRefGenerator, IrTypeStructRef);
` and thus on other types, I think it's not worth it

Reply:
AFAIK if we do not want to expose it in other places, maybe we can write an *extra* impl block, without worrying about other `impl` blocks. Rust's grammar is expressive in this situation imho.

```rust
impl TypeStructRefGenerator {
  fn dart_implements(&self) { ... }
}

// btw if you want you can even write down more impl
impl TypeStructRefGenerator {fn other_func(){}}
```